samples/cpp/text2image/main.cpp

```diff
@@ -12,11 +12,18 @@ int32_t main(int32_t argc, char* argv[]) try {
     const std::string device = "CPU";  // GPU, NPU can be used as well
 
     ov::genai::Text2ImagePipeline pipe(models_path, device);
+
+    auto callback = [&](size_t step, ov::Tensor& intermediate_res) -> bool {
+        std::cout << "Image generation step: " << step << std::endl;
+        return false;
+    };
```

Comment:
maybe we can keep samples as is and move such snippet to documentation of the `ov::genai::callback` property?

Otherwise, samples are overcomplicated with a lot of non-required parameters, while they should be as simple as possible.